src/figanos/matplotlib/plot.py

```diff
@@ -1409,21 +1431,57 @@ def heatmap(
     plot_kw.setdefault("cmap", cmap)
 
     # plot
-    sns.heatmap(df, ax=ax, **plot_kw)
-
-    # format
-    plt.xticks(rotation=45, ha="right", rotation_mode="anchor")
-    ax.tick_params(axis="both", direction="out")
+    def draw_heatmap(*args, **kwargs):
+        data = kwargs.pop("data")
+        d = (
+            data
+            if len(args) == 0
+            else data.pivot(index=args[1], columns=args[0], values=args[2])
+        )
+        ax = sns.heatmap(d, **kwargs)
+        ax.set_xticklabels(
+            ax.get_xticklabels(), rotation=45, ha="right", rotation_mode="anchor"
+        )
+        ax.tick_params(axis="both", direction="out")
+        set_plot_attrs(
+            use_attrs,
+            da,
+            ax,
+            title_loc="center",
+            wrap_kw={"min_line_len": 35, "max_line_len": 44},
+        )
+        return ax
 
-    set_plot_attrs(
-        use_attrs,
-        da,
-        ax,
-        title_loc="center",
-        wrap_kw={"min_line_len": 35, "max_line_len": 44},
-    )
+    if ax is not None:
+        ax = draw_heatmap(data=df, ax=ax, **plot_kw)
+        return ax
+    elif "col" in plot_kw or "row" in plot_kw:
+        # When using xarray's FacetGrid, `plot_kw` can be used in the FacetGrid and in the plotting function
+        # With Seaborn, we need to be more careful and separate keywords.
+        plot_kw_hm = {
+            k: v for k, v in plot_kw.items() if k in signature(sns.heatmap).parameters
+        }
+        plot_kw_fg = {
+            k: v for k, v in plot_kw.items() if k in signature(sns.FacetGrid).parameters
+        }
+        unused_keys = (
+            set(plot_kw.keys()) - set(plot_kw_fg.keys()) - set(plot_kw_hm.keys())
+        )
+        if unused_keys != set():
+            raise ValueError(
+                f"`heatmap` got unexpected keywords in `plot_kw`: {unused_keys}. Keywords in `plot_kw` should be keywords "
+                "allowed in `sns.heatmap` or `sns.FacetGrid`. "
+            )
```

Comment:
Just want to highlight this: This ensures we can use both keywords from heatmap and FacetGrid. With this approach, we would silently ignore wrong keywords, so I have to add an extra step to throw an error. 

A solution to avoid this extra step would be to stuff in illegal keywords in `plot_kw_fg` for instance, then `sns.FacetGrid` would handle the error management. But since `plot_kw` allows for both keywords on FacetGrid and Heatmap, I thought it was better to have our own explicit error message, since we have this peculiar input.